src/dispatch/auth/views.py

```diff
@@ -145,10 +145,13 @@ def update_user(
         )
 
     if user_in.role:
```

Comment:
Permission check logic was moved to the function to only check for owner role when role change is requested and still allow for other changes like default projects.